src/modules/dashboard/components/HookedTokenIcon/HookedTokenIcon.tsx

```diff
@@ -24,76 +28,63 @@ interface Props {
 
   /** Optional name for the icon title */
   name?: string;
-}
 
-const checkSVG = (fileName: string) =>
-  fileName &&
-  fileName.substring(fileName.length - 3, fileName.length) === 'svg';
+  /** If provided than icon is display instead of Avatar */
+  iconName?: string;
 
-const loadTokenImages = async (logo): Promise<ImageType> =>
-  import(
-    /* webpackMode: "eager" */ `../../../../../node_modules/eth-contract-metadata/images/${logo}`
-  );
+  /** If true logo fetching wont be fire */
+  dontFetch?: boolean;
+}
+
+const loadTokenImages = async (address: Address): Promise<Response> => {
+  let tokenImageUrl = `${TOKEN_LOGOS_REPO_URL}${address}/logo.png`;
+  if (address === AddressZero) {
+    tokenImageUrl = `${TOKEN_LOGOS_REPO_URL}info/logo.png`;
+  }
+  return fetch(tokenImageUrl);
+};
```

Comment:
I'm all good with this, but I want to add more functionality. Sorry :)

We need an internal state that tracks if the icon `isFetching` and based on that show a loading spinner over the avatar. _(but only the avatar, there's no point in having it, if we just show a manual icon)_

This is because there's a flash of content while fetching, and when replacing the token avatar _(plus the user doesn't actually know that we're fetching in the background)_

![demo-slow-token-loading](https://user-images.githubusercontent.com/1193222/101896560-d85a9f80-3bb1-11eb-89c2-33ebff184fee.gif)

We have something similar for the `AvatarUploader` core component, _if you need_ to take inspiration from somewhere: https://github.com/JoinColony/colonyDapp/blob/master/src/modules/core/components/AvatarUploader/AvatarUploadItem.tsx#L79-L81

Otherwise, just implement this as you see fit :smile:

Reply:
The way I'm seeing this look is an overlay over the existing blockie (just show the blockie initially), and that overlay has a small spinner loader inside of it.

When the file has finished loading, replace the blockie in the background (with the overlay still active), then remove the overlay entirely.

This way we both show a loading state to the user and prevent having the content flash white in the interim state after the file has been fetched and is being replaced.